src/main/webapp/app/exercises/shared/exercise/exercise.component.ts

```diff
@@ -57,14 +62,18 @@ export abstract class ExerciseComponent implements OnInit, OnDestroy {
             this.loadCourse();
         } else {
             this.courseId = this.course.id;
-            this.loadExercises();
+            if (this.isInExerciseGroup == null) {
+                this.loadExercises();
+            }
         }
     }
 
     private loadCourse(): void {
         this.courseService.find(this.courseId).subscribe((courseResponse) => {
             this.course = courseResponse.body!;
-            this.loadExercises();
+            if (this.isInExerciseGroup == null) {
```

Comment:
For example: its more readable if this would be `if(this.isInExerciseGroup)`

Reply:
I made the change accordingly